### PR TITLE
feat: levearge for loop to init `poolAssets` element and use solidity nomenclature

### PR DESCRIPTION
**Pull request by petrovska-petro**

### CODE REVIEW

**Review by gosuto-inzasheru — DISMISSED** [review]

**Comment by petrovska-petro** [comment]
decided to included in the same pr also the precision tightening assert we chatted about instead of making separate pr w/ the minor commit at 3cc8463ad81dccbe50c44ebe05ea9e7a2f9875fa

File: test/TopupEUReTest.t.sol [hunk 1]
```diff
@@ -66,29 +66,33 @@ contract TopupTest is BaseFixture {
...
         // ensure default values at `txQueueData` after execution
         _assertPostDefaultValuesNextTxExec();
 
-        assertLt(
+        assertApproxEqAbs(
             IERC20(BPT_STEUR_EURE).balanceOf(GNOSIS_SAFE),
-            initialBptBal,
+            initialBptBal - maxBPTAmountIn,
+            DIFF_MIN_OUT_CALC_ALLOWED,
             "BPT balance: not decreased after withdrawing from the pool"
```
Comment (gosuto-inzasheru) [inline test/TopupEUReTest.t.sol:90]: idem

File: test/TopupEUReTest.t.sol [hunk 2]
```diff
@@ -66,29 +66,33 @@ contract TopupTest is BaseFixture {
...
         );
-        assertGt(
+
+        assertEq(
             IERC20(EURE).balanceOf(GNOSIS_SAFE),
-            initialEureBal,
+            initialEureBal + _deficit,
             "EURE balance: not increased after withdrawing from the pool"
```
Comment (gosuto-inzasheru) [inline test/TopupEUReTest.t.sol:96]: also not accurate description?

File: test/TopupBptTest.t.sol [hunk 3]
```diff
@@ -73,14 +75,15 @@ contract TopupBptTest is BaseFixture {
...
+            initialEureBal - _amount,
             "EURE balance: did not decrease after depositing into the pool"
         );
-        assertGt(
+        assertApproxEqAbs(
             IERC20(BPT_STEUR_EURE).balanceOf(GNOSIS_SAFE),
-            initialBptBal,
+            initialBptBal + bptOutExpected,
+            DIFF_MIN_OUT_CALC_ALLOWED,
             "BPT balance: did not increase after depositing into the pool"
```
[the remark this answers is not in the document]
Comment (petrovska-petro) [reply test/TopupBptTest.t.sol]: ad536a7b16c5679d767dd165cabe610577ad87f4

File: test/TopupEUReTest.t.sol (hunk 1 above)
Comment (petrovska-petro) [reply test/TopupEUReTest.t.sol]: ad536a7b16c5679d767dd165cabe610577ad87f4

File: test/TopupEUReTest.t.sol (hunk 2 above)
Comment (petrovska-petro) [reply test/TopupEUReTest.t.sol]: ad536a7b16c5679d767dd165cabe610577ad87f4

**Review by gosuto-inzasheru — APPROVED** [review]

**Comment by gosuto-inzasheru** [comment]
good to see these passing, especially the `assertApproxEqAbs`

**Pull request merged**